### PR TITLE
Fixed Fictional Weapons mod (tested offline this time)

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
Per community requirements, I have tested this version offline before opening another PR, and I did not combine formatting changes with code changes.

As I said in my PM, I apologize for the low quality of my previous PRs. I will not repeat that.

From #18801 

> Several things were wrong with the mod: the Five0 Handcannon could not be used due to a range penalty of -70; The arc cannon had a low damage, several modslots didn't make sense; all the guns had ridiculous loudness penalties; etc..

EDIT: For those interested, you need to add ~1 to the end of your rebase -i to include the base file.i used it to clean the botched rebase. Thank you for the help, Coolthulhu.

The MP5 EOD was replaced by the Eagle 1776 .44 Magnum smg, and the Rico 454 was replaced by the Boomlighter, which has a flamer attachment.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
Can migration files be used inside mods?

**Comment by MystiaLorelei** [comment]
Just going to drop in here to answer something from the earlier PR. I'm certain that MP5 variant is indeed based on the one from Half-Life. It's appearance in that Arma 3 RPG mod\server is because another modder created an Arma 3 mp5 pack and included that as a variant amongst the (many) real world ones.

http://www.armaholic.com/datas/users/niarms-mp5-smgs-version-15-6_4.jpg the preview image for it makes it pretty obvious. 

It's replacement looks pretty cool.

**Comment by mugling** [comment]
> Can migration files be used inside mods?

Yes, but you shouldn't do this as you cannot guarantee what other mods will also be loaded.

Indeed now I think about it I'm probably going to open a PR disable loading of `MIGRATION` for mods

**Comment by mugling** [comment]
> Per community requirements, I have tested this version offline before opening another PR, and I did not combine formatting changes with code changes.
> 
> As I said in my PM, I apologize for the low quality of my previous PRs. I will not repeat that.

Thank you, this is appreciated

**Comment by BorkBorkGoesTheCode** [comment]
>Indeed now I think about it I'm probably going to open a PR disable loading of  MIGRATION  for mods

Please don't. I changed the item id of two guns in this PR, and I don't want to cause broken items.

**Comment by mugling** [comment]
Don't change the `id`

**Comment by Zireael07** [comment]
@mugling: 
The OP states:
>The MP5 EOD was replaced by the Eagle 1776 .44 Magnum smg, and the Rico 454 was replaced by the Boomlighter, which has a flamer attachment.

Therefore the id's need to be changed as the old ids have zilch to do with the new weapons.

**Comment by BorkBorkGoesTheCode** [comment]
>Don't change the  id 

The old names are from Half-Life and Just Cause.

**Comment by mugling** [comment]
Think of the `id` as an arbitrary key. It doesn't have to relate to the `item`. I tried for similar changes previously and @BevapDin pointed out this just adds complexity for no good cause.

**Comment by BorkBorkGoesTheCode** [comment]
>Think of the  id  as an arbitrary key. It doesn't have to relate to the  item . I tried for similar changes previously and @BevapDin pointed out this just adds complexity for no good cause.

I know the id is an arbitrary thing, they still need to be changed:

>Who included the MP5A5 EOD? It looks like it's from City LifeRPG, maybe Half-Life.

>Just going to drop in here to answer something from the earlier PR. I'm certain that MP5 variant is indeed based on the one from Half-Life. It's appearance in that Arma 3 RPG mod\server is because another modder created an Arma 3 mp5 pack and included that as a variant amongst the (many) real world ones.

http://www.armaholic.com/datas/users/niarms-mp5-smgs-version-15-6_4.jpg the preview image for it makes it pretty obvious.

**Comment by mugling** [comment]
> I know the id is an arbitrary thing, they still need to be changed:

Why? Guaranteed exciting breakage if two mods try and migrate the same item? Too many interactions, blacklists, whitelists, legacy savegames, core migrations, mods...

**Comment by BorkBorkGoesTheCode** [comment]
>Why? Guaranteed exciting breakage if two mods try and migrate the same item? Too many interactions, blacklists, whitelists, legacy savegames, core migrations, mods...

These ids are unique to this mod; setting up a migration will not cause significant damage, unless mod migration isn't supported yet.

**Comment by mugling** [comment]
Why do you need to migrate the `id` though? I'm not disagreeing it would be nice, but note the word **need*. We're potentially enabling a lot of problems in the future by allowing mods to do that

File: data/mods/FictonalWeapons/fic_weapons.json [hunk 1]
```diff
@@ -1,30 +1,29 @@
 [
   {
-    "id": "ficrico",
+    "id": "boomlighter",
```
Comment (mugling) [inline data/mods/FictonalWeapons/fic_weapons.json]: Break in savegame compatibility

**Comment by Coolthulhu** [comment]
I don't know what to do here, I'm deassigning.

I'd allow migration here until 0.D. I don't think any mod inherits this one, so while not ideal to migrate in mods, it should cause any problems.

This is a relatively unupdated mod and it's better to cause some chaos than to leave it rusting.

File: data/mods/FictonalWeapons/fic_weapons.json (hunk 1 above)
Comment (mugling) [inline data/mods/FictonalWeapons/fic_weapons.json]: Just use the original `id`

**Pull request closed without merging**